### ISSUE TITLE
Segmentation fault of arp-scan on Ubuntu 17.04

### ISSUE THREAD

**Issue by kp625544** [royhills/arp-scan#27]
Hello team,
I have found a issue with the arp-scan which may potentially be executed for a buffer overflow attack.
Whenever I run arp-scan on ubuntu 17.04 it crases with segmentation fault and with a sigsegv fault.
Thank you

**Comment by kp625544** [comment]
Here is an unlisted YouTube video regarding the same.
https://youtu.be/_lDmscVBZbE

**Comment by royhills** [comment]
Thanks for the bug report.

The ubuntu package arp-scan_1.9-1_amd64.deb appears to be based on a version of arp-scan that's about four years old.  If you look in /usr/share/arp-scan/ieee-oui.txt you'll see:

# This file was automatically generated by get-oui at 2013-07-24 09:38:43

There have been quite a few upstream changes since then, so I'd suggest pulling the latest version from github and trying that.  If that doesn't exhibit this segfault behaviour, then you might want to contact the ubuntu package maintainer and suggest that they update the package from github.

**Comment by royhills** [comment]
If you look at the arp-scan changelog file in the ubuntu package, and compare it with the latest changelog file on github, you can work out what's changed between the two versions.

**Comment by kp625544** [comment]
Ok I ll try doing the same and also contacting upstream to update the repo. Will get back here after it's done.

**Comment by bcoles** [comment]
I'm not sure about Ubuntu 17, but this is resolved on Ubuntu 20.04.2:

```
# arp-scan --version
arp-scan 1.9.7

Copyright (C) 2005-2019 Roy Hills, NTA Monitor Ltd.
arp-scan comes with NO WARRANTY to the extent permitted by law.
You may redistribute copies of arp-scan under the terms of the GNU
General Public License.
For more information about these matters, see the file named COPYING.

libpcap version 1.9.1 (with TPACKET_V3)
```

```
$ apt info arp-scan
Package: arp-scan
Version: 1.9.7-1
Priority: extra
Section: universe/admin
Origin: Ubuntu
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Debian Security Tools <team+pkg-security@tracker.debian.org>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 1,095 kB
Depends: libc6 (>= 2.15), libpcap0.8 (>= 1.5.1), ieee-data
Recommends: libwww-perl
Homepage: https://github.com/royhills/arp-scan
Download-Size: 338 kB
APT-Manual-Installed: yes
APT-Sources: http://us.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
Description: arp scanning and fingerprinting tool
 arp-scan is a command-line tool that uses the ARP protocol to discover and
 fingerprint IP hosts on the local network. It is available for Linux and BSD
 under the GPL licence
```

```
$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.2 LTS"
```

```
$ grep generated /usr/share/arp-scan/ieee-oui.txt
# This file was automatically generated by get-oui at 2019-10-13 09:23:00
```

**Comment by royhills** [comment]
I'm closing this issue as I beleive the problem has been addressed in the latest git release.  Please re-open if you think the issue has not been solved. Thanks.
